host/host.go

```diff
@@ -0,0 +1,69 @@
+// Package host provides facilities for interfacing with the host operating
+// system, logging activities performed.
+package host
+
+import (
+	"log"
+	"os/exec"
+	"strings"
+
+	"github.com/taskcluster/shell"
+)
+
+// Run is equivalent to exec.Command(name, arg...).Run() but with logging.
+func Run(name string, arg ...string) (err error) {
+	_, err = RunCommand(exec.Command(name, arg...))
+	return
+}
+
+// CombinedOutput is equivalent to exec.Command(name, arg...).CombinedOutput()
+// but with logging.
+func CombinedOutput(name string, arg ...string) (combinedOutput string, err error) {
+	return RunCommand(exec.Command(name, arg...))
+}
+
+// RunBatch calls Run for each command in commands, in sequence. If allowFail
+// is false it will return if an error is returned from Run. All errors are
+// logged regardless of allowFail. The returned error is the result of the last
+// Run call.
+func RunBatch(allowFail bool, commands ...[]string) (err error) {
```

Comment:
The normal flow for `RunBatch` will double print output in the case of an error.

`RunBatch()` (prints errors) => `Run()` => `RunCommand()` (prints errors)

Reply:
No, `RunCommand()` logs the output of the failed command, whereas RunBatch() displays the (go) error (e.g. 'exit code 65'). The `RunCommand` docs explain that the caller should log the error, and it will log the command output.